src/DatePicker/DatePicker.js

```diff
@@ -206,13 +209,55 @@ class DatePicker extends Component {
         date: date,
       });
     }
+
+    setTimeout(() => {
+      this.setState({
+        isFocused: false,
+      });
+    }, 0);
+    this.refs.input.focus();
+
     if (this.props.onChange) {
       this.props.onChange(null, date);
     }
   };
 
+  handleBlur = (event, shouldSetFocus) => {
+    if (shouldSetFocus) {
+      this.refs.input.focus();
+    }
+
+    if (this.state.isFocused) {
+      this.setState({
+        isFocused: false,
+      }, this.refs.dialogWindow.dismiss);
+    }
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  };
+
+  handleDismiss = () => {
+    if (this.state.isFocused) {
+      this.refs.input.focus();
+      this.setState({
+        isFocused: false,
+      });
+    }
+
+    if (this.props.onDismiss) {
+      this.props.onDismiss();
+    }
+  };
+
   handleFocus = (event) => {
-    event.target.blur();
+    if (!this.props.disabled && !this.state.isFocused) {
```

Comment:
@oliviertassinari  ah, sorry, the change was added. I wasn't meant to commit it, a long few days :| 

I can leave it changed and add the `onTouchTap` method with a note in the documents about it being deprecated, undo what I did or whatever you think is best